scripts/import_libcbor.py

```diff
@@ -0,0 +1,116 @@
+import os
+import tempfile
+import shutil
+import subprocess
+import re
+
+# TODO support pass-in version and use the tag
+
+major_version = 0
+minor_version = 11
+patch_version = 0
+
+GENERATED_NOTES = """/**
+ * DO NOT DIRECTLY MODIFY THIS FILE:
+ *
+ * The code in this file is generated from scripts/import_libcbor.py
+ *  and any modifications should be in there.
+ */
+"""
+
+CBOR_EXPORT_H = """
+#ifndef CBOR_EXPORT_H
+#define CBOR_EXPORT_H
+
+/* Don't export anything from libcbor */
+#define CBOR_EXPORT
+
+#endif /* CBOR_EXPORT_H */
+"""
+
+CONFIGURATION_H = f"""
+#ifndef LIBCBOR_CONFIGURATION_H
+#define LIBCBOR_CONFIGURATION_H
+
+#define CBOR_MAJOR_VERSION {major_version}
+#define CBOR_MINOR_VERSION {minor_version}
+#define CBOR_PATCH_VERSION {patch_version}
+
+#define CBOR_BUFFER_GROWTH 2
+#define CBOR_MAX_STACK_SIZE 2048
+#define CBOR_PRETTY_PRINTER 1
+
+#if defined(_MSC_VER)
+#    define CBOR_RESTRICT_SPECIFIER
+#else
+#    define CBOR_RESTRICT_SPECIFIER restrict
+#endif
+
+#define CBOR_INLINE_SPECIFIER
+
+/* Ignore the compiler warnings for libcbor. */
+#ifdef _MSC_VER
+#    pragma warning(disable : 4028)
+#    pragma warning(disable : 4715)
+#    pragma warning(disable : 4232)
+#    pragma warning(disable : 4068)
+#    pragma warning(disable : 4244)
+#endif
+
+#ifdef __clang__
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wreturn-type"
+#elif defined(__GNUC__)
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wreturn-type"
+#    pragma GCC diagnostic ignored "-Wunknown-pragmas"
+#endif
+
+#endif // LIBCBOR_CONFIGURATION_H
+"""
+
+# Create a temporary directory for cloning the repository
+temp_repo_dir = tempfile.mkdtemp()
+
+try:
+    # Clone the repository into the temporary directory
+    repo_url = "https://github.com/PJK/libcbor.git"
+    clone_command = f"git clone {repo_url} {temp_repo_dir}"
+    subprocess.run(clone_command, shell=True, check=True)
+
+    # Create a separate folder for the copied files
+    output_dir = os.path.join(
+        os.path.dirname(__file__), "..", "source", "external", "libcbor")
+    os.makedirs(output_dir, exist_ok=True)
```

Comment:
delete everything in output_dir before copying things in again

that way, if libcbor ends up renaming or removing a file, we'll also get rid of the old file